Library/Homebrew/formulary.rb

```diff
@@ -160,17 +160,17 @@ def initialize tapped_name
       @tapped_name = tapped_name
       user, repo, name = tapped_name.split("/", 3).map(&:downcase)
       tap = Pathname.new("#{HOMEBREW_LIBRARY}/Taps/#{user}/homebrew-#{repo}")
-      path = tap.join("#{name}.rb")
 
       if tap.directory?
         tap.find_formula do |file|
           if file.basename(".rb").to_s == name
-            path = file
+            super name, file
+            return
           end
         end
       end
 
-      super name, path
+      raise TapFormulaUnavailableError, tapped_name
     end
```

Comment:
What I am trying to do is to make sure either we raise error and the formula has been found. Without this line, we may get a state that formula isn't found and won't raise error unless we call `get_formula`. This creates a lot of headache. We have to do a second file checking in the downstream. (There are times that we need check formula existing without the need to read actual formula, like `edit` and `log`). 

So IMHO, this makes things simpler.